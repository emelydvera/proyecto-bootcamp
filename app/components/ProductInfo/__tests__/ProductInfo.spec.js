const React = require("react");
const ProductInfo = require("../ProductInfo");
const { render, screen } = require("@testing-library/react");
const {
  data,
} = require("../../../../mocks/test/get/https/api.mercadolibre.com/items/MLA1104983845.json");

describe("must show product information", () => {
  it("matches snapshot", () => {
    const props = {
      product: data,
      quantity: "2",
      amount: "1200",
    };
    const { asFragment } = render(<ProductInfo {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("product condition is new", () => {
    const props = {
      product: data,
      quantity: "2",
      amount: "1200",
    };

    render(<ProductInfo {...props} />);
    const span = document.querySelector(".info__use");
    expect(span).toBeInTheDocument();
    expect(span).toHaveTextContent("Nuevo");
  });

  it("the product is not specified", () => {
    const props = {
      product: { ...data, condition: "not_specified" },
      quantity: "2",
      amount: "1200",
    };
    render(<ProductInfo {...props} />);
    const span = document.querySelector(".info__use");
    expect(span).toBeInTheDocument();
    expect(span).toHaveTextContent("No Especifica");
  });

  it("product condition is used", () => {
    const props = {
      product: { ...data, condition: "used" },
      quantity: "2",
      amount: "1200",
    };
    render(<ProductInfo {...props} />);
    const span = document.querySelector(".info__use");
    expect(span).toBeInTheDocument();
    expect(span).toHaveTextContent("Usado");
  });

  it("product condition is empty", () => {
    const props = {
      product: { ...data, condition: "" },
      quantity: "2",
      amount: "1200",
    };
    render(<ProductInfo {...props} />);
    const span = document.querySelector(".info__use");
    expect(span).toBeInTheDocument();
    expect(span).toHaveTextContent("");
  });
});
