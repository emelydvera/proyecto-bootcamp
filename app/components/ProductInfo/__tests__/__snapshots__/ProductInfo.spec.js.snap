// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must show product information matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="info"
  >
    <span
      aria-label="Nuevo"
      class="info__use"
      tabindex="10"
    >
      Nuevo
    </span>
    <span
      aria-label="150 productos vendidos"
      class="info__sold"
      tabindex="11"
    >
       - 150 vendidos
    </span>
    <h2
      aria-label=" Moto G20 64 Gb  Azul Cielo 4 Gb Ram"
      tabindex="12"
    >
       Moto G20 64 Gb  Azul Cielo 4 Gb Ram
    </h2>
    <div
      tabindex="13"
    >
      <span
        class="andes-money-amount info__buy__price andes-money-amount--cents-superscript"
        style="font-size: 48px;"
      >
        <span
          class="andes-visually-hidden"
        >
          30999 pesos
        </span>
        <span
          aria-hidden="true"
          class="andes-money-amount__currency-symbol"
        >
          $
        </span>
        <span
          aria-hidden="true"
          class="andes-money-amount__fraction"
        >
          30.999
        </span>
        <span
          aria-hidden="true"
          class="andes-visually-hidden"
        >
          ,
        </span>
        <span
          aria-hidden="true"
          class="andes-money-amount__cents andes-money-amount__cents--superscript-48"
          style="font-size: 24px; margin-top: 9px;"
        >
          00
        </span>
      </span>
    </div>
    <div
      aria-label="Pagalo en $2 cuotas, cada una de  $1200"
      class="info__buy__quantity-amount"
      tabindex="14"
    >
      2 cuotas de 
      <span
        class="andes-money-amount info__buy__amount andes-money-amount--cents-superscript"
        style="font-size: 16px;"
      >
        <span
          class="andes-visually-hidden"
        >
          1200 pesos
        </span>
        <span
          aria-hidden="true"
          class="andes-money-amount__currency-symbol"
        >
          $
        </span>
        <span
          aria-hidden="true"
          class="andes-money-amount__fraction"
        >
          1.200
        </span>
        <span
          aria-hidden="true"
          class="andes-visually-hidden"
        >
          ,
        </span>
        <span
          aria-hidden="true"
          class="andes-money-amount__cents andes-money-amount__cents--superscript-16"
          style="font-size: 10px; margin-top: 2px;"
        >
          00
        </span>
      </span>
       cada una
    </div>
    <p
      aria-label="hay $1 unidades disponibles"
      class="info__available"
      tabindex="16"
    >
      Cantidad disponible: 1
    </p>
    <div
      class="info__shipping"
    >
      <svg
        fill=" #00A650"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          clip-rule="evenodd"
          d="M13.499 3.75H3C1.75736 3.75 0.75 4.75736 0.75 6V16.4995C0.75 17.7421 1.75736 18.7495 3 18.7495H4.5563C4.89024 20.0646 6.08201 21.0375 7.501 21.0375C8.91999 21.0375 10.1118 20.0646 10.4457 18.7495H14.2643C14.5983 20.0646 15.79 21.0375 17.209 21.0375C18.628 21.0375 19.8198 20.0646 20.1537 18.7495H23.25V11.7708L19.8654 6.75H15.749V6C15.749 4.75736 14.7416 3.75 13.499 3.75ZM10.4457 17.2495H14.249V6C14.249 5.58579 13.9132 5.25 13.499 5.25H3C2.58579 5.25 2.25 5.58579 2.25 6V16.4995C2.25 16.9137 2.58579 17.2495 3 17.2495H4.5563C4.89024 15.9344 6.08201 14.9615 7.501 14.9615C8.91999 14.9615 10.1118 15.9344 10.4457 17.2495ZM17.209 14.9615C16.6803 14.9615 16.183 15.0966 15.75 15.3342V8.25H19.0676L21.75 12.2292V17.2495H20.1537C19.8198 15.9344 18.628 14.9615 17.209 14.9615ZM15.671 17.9995C15.671 18.8489 16.3596 19.5375 17.209 19.5375C18.0584 19.5375 18.747 18.8489 18.747 17.9995C18.747 17.1501 18.0584 16.4615 17.209 16.4615C16.3596 16.4615 15.671 17.1501 15.671 17.9995ZM7.501 19.5375C6.6516 19.5375 5.96302 18.8489 5.96302 17.9995C5.96302 17.1501 6.6516 16.4615 7.501 16.4615C8.3504 16.4615 9.03898 17.1501 9.03898 17.9995C9.03898 18.8489 8.3504 19.5375 7.501 19.5375Z"
          fill=" #00A650"
          fill-rule="evenodd"
        />
      </svg>
      <p
        aria-label="el envio es gratis"
        tabindex="17"
      >
        Envío gratis
      </p>
    </div>
    <div
      class="buy__container"
    >
      <div
        class="input__quantity"
      >
        <div
          class="andes-form-control andes-form-control--textfield andes-form-control--centered andes-form-control--default buy__input"
        >
          <label>
            <span
              class="andes-form-control__label"
            >
              Cantidad
            </span>
            <div
              class="andes-form-control__control"
            >
              <input
                aria-label="Ingrese la cantidad de productos a comprar"
                class="andes-form-control__field"
                maxlength="120"
                rows="1"
                tabindex="18"
                type="number"
                value="1"
              />
            </div>
          </label>
          <div
            class="andes-form-control__bottom"
          >
            <span
              class="andes-form-control__message"
            />
          </div>
        </div>
      </div>
      <button
        aria-label="botón para comprar producto"
        class="andes-button buy__button andes-button--large andes-button--loud andes-button--full-width"
        tabindex="18"
        type="submit"
      >
        <span
          class="andes-button__content"
        >
          Comprar
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
